### PR TITLE
fix: remove period from service name pattern validation

### PR DESCRIPTION
**Pull request by agatha197**
**Changes:**

This PR updates the service name validation rule across multiple language files and the React component. The changes include:

1. Modifying the regex pattern in `ServiceLauncherPageContent.tsx` to enforce stricter rules for service names:
   - Minimum length of 4 characters
   - Maximum length of 24 characters
   - Cannot start or end with a hyphen
   - Only alphanumeric characters, underscores, and hyphens are allowed

2. Updating the `ServiceNameRule` translation string in all language files to reflect the new rule.

3. Adding new translation strings for minimum length, maximum length, and hyphen placement rules.

**Rationale:**

The updated service name validation rules aim to improve consistency and prevent potential issues with service name parsing or conflicts with various systems. The stricter rules ensure more standardized and predictable service names across the platform.

**Impact:**

- Users will need to follow the new, more specific rules when creating or modifying service names.
- Existing services with names that don't comply with the new rules may need to be renamed.
- The user interface will provide more detailed feedback on service name requirements.

**Checklist:**

- [ ] Documentation: Update any relevant documentation on service naming conventions
- [ ] Test case(s): Add tests to verify that service names comply with the new rules, including length restrictions and character placement
- [ ] UI/UX: Ensure that the new validation messages are displayed correctly and provide clear guidance to users

### CODE REVIEW

**Comment by agatha197** [comment]
* **#2773** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2773?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @agatha197 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/ServiceLauncherPageContent.tsx [hunk 1]
```diff
@@ -726,7 +726,7 @@ const ServiceLauncherPageContent: React.FC<ServiceLauncherPageContentProps> = ({
                         name="serviceName"
                         rules={[
                           {
-                            pattern: /^(?=.{4,24}$)\w[\w.-]*\w$/,
+                            pattern: /^(?=.{4,24}$)[\w-]+$/,
```
Comment (inureyes) [inline react/src/components/ServiceLauncherPageContent.tsx]: Let's change the pattern to /^(?=.{4,24}$)\w[\w-]*\w$/
(disallow strings that starts/ ends with hyphen-)

File: react/src/components/ServiceLauncherPageContent.tsx [hunk 2]
```diff
@@ -726,7 +726,7 @@ const ServiceLauncherPageContent: React.FC<ServiceLauncherPageContentProps> = ({
                         name="serviceName"
                         rules={[
                           {
-                            pattern: /^(?=.{4,24}$)\w[\w.-]*\w$/,
+                            pattern: /^(?=.{4,24}$)\w[\w-]*\w$/,
```
Comment (yomybaby) [inline react/src/components/ServiceLauncherPageContent.tsx]: Let's separate the rule into 3 rules:

- Minimum
- Maximum
- Allowed characters

Without this separation, users will receive a long validation error message when typing less than 4 characters, even if it doesn't have any non-allowed characters.

**Review by yomybaby — CHANGES_REQUESTED** [review]

File: react/src/components/ServiceLauncherPageContent.tsx (hunk 2 above)
Comment (agatha197) [reply react/src/components/ServiceLauncherPageContent.tsx]: I have separated one more rule. Start/end with a hyphen.

**Review by yomybaby — APPROVED** [review]
LGTM

**Pull request merged**